agent/asm/asm_test.go

```diff
@@ -111,11 +121,74 @@ func TestASMGetAuthConfig(t *testing.T) {
 }
 
 func TestGetSecretFromASM(t *testing.T) {
-	asmClient := mockGetSecretValue{
-		Resp: secretsmanager.GetSecretValueOutput{
-			SecretString: aws.String("secretValue"),
-		},
-	}
+	asmClient := createASMInterface(secretValue)
 	_, err := GetSecretFromASM("secretName", asmClient)
 	assert.NoError(t, err)
 }
+
+func TestGetSecretFromASMWithJsonKey(t *testing.T) {
+	asmClient := createASMInterface(jsonSecretValue)
+	secretValueInput := createSecretValueInput(toPtr(valueFrom), nil, nil)
+	outSecretValue, _ := GetSecretFromASMWithInput(secretValueInput, asmClient, jsonKey)
+	assert.Equal(t, secretValue, outSecretValue)
+}
+
+func TestGetSecretFromASMWithMalformedJSON(t *testing.T) {
+	asmClient := createASMInterface(malformedJsonSecretValue)
+	secretValueInput := createSecretValueInput(toPtr(valueFrom), nil, nil)
+	outSecretValue, err := GetSecretFromASMWithInput(secretValueInput, asmClient, jsonKey)
+	if assert.Error(t, err) {
```

Comment:
i think you can just use require to avoid the need of this kind of check, i.e:
```
require.Error(...)
assert.Equal(...)
```
Same below

Reply:
i did it this way bc this was the examples in assert pkg documentation. is require better in this case? they seem to both accomplish the same things. i can change it to keep with our existing coding conventions.

Reply:
non blocking but would be good to keep with convention

Reply:
changed